vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -528,6 +530,134 @@ public boolean isResizable() {
         return getElement().getProperty("resizable", false);
     }
 
+    /**
+     * Sets the title to be rendered on the dialog header.
+     *
+     * @param title
+     *            title to be rendered
+     */
+    public void setHeaderTitle(String title) {
+        getElement().setProperty("headerTitle", title);
+    }
+
+    /**
+     * Gets the title set for the dialog header.
+     *
+     * @return the title or an empty string, if a header title is not defined.
+     */
+    public String getHeaderTitle() {
+        return getElement().getProperty("headerTitle", "");
+    }
+
+    /**
+     * Gets the object from which components can be added or removed from the
+     * dialog header area. The header is displayed only if there's a
+     * {@link #getHeaderTitle()} or at least one component added with
+     * {@link DialogHeaderFooter#add(Component)}.
+     *
+     * @return the header object
+     */
+    public DialogHeader getHeader() {
+        if (this.dialogHeader == null) {
+            this.dialogHeader = new DialogHeader(this);
+        }
+        return this.dialogHeader;
+    }
+
+    /**
+     * Gets the object from which components can be added or removed from the
+     * dialog footer area. The footer is displayed only if there's at least one
+     * component added with {@link DialogHeaderFooter#add(Component)}.
+     *
+     * @return the header object
+     */
+    public DialogFooter getFooter() {
+        if (this.dialogFooter == null) {
+            this.dialogFooter = new DialogFooter(this);
+        }
+        return this.dialogFooter;
+    }
+
+    public static class DialogHeader extends DialogHeaderFooter {
+        private DialogHeader(Dialog dialog) {
+            super("headerRenderer", dialog);
+        }
+    }
+
+    public static class DialogFooter extends DialogHeaderFooter {
+        private DialogFooter(Dialog dialog) {
+            super("footerRenderer", dialog);
+            root.getStyle().set("flex", "1");
+            root.getStyle().set("justify-content", "flex-end");
+        }
+    }
+
+    static abstract class DialogHeaderFooter {
```

Comment:
Should this extend Element or Component to allow people to interact with the underlying root container instead of adding it inside and can't be changed later on? This would allow to add simple css through the java API to the root element or traverse the children of it. Now it's impossible to get the children of the header or footer

Reply:
I first thought of making it extend `Element` but my fear would be that we would have to support the whole API it brings with it. I went with this approach so that we can decide which `Element` (or `Component`) API to support on header and footer. For instance, we could add `getStyle()` method that would simply return `root.getStyle()`and so on.

But, of course, this is up for debate.